marshal.go

```diff
@@ -908,71 +885,58 @@ func marshalVarbind(pdu *SnmpPDU) ([]byte, error) {
 		intBuf := new(bytes.Buffer)
 		intBuf.WriteByte(byte(AsnExtensionTag))
 		intBuf.WriteByte(byte(pdu.Type))
-		intBytes, err := converters[pdu.Type](pdu.Value)
-		if err != nil {
-			return nil, fmt.Errorf("error converting PDU value type %v to %v: %w", pdu.Value, pdu.Type, err)
+		intBytes, tlvErr := converters[pdu.Type](pdu.Value)
+		if tlvErr != nil {
+			return nil, fmt.Errorf("error converting PDU value type %v to %v: %w", pdu.Value, pdu.Type, tlvErr)
 		}
-		intLength, err := marshalLength(len(intBytes))
-		if err != nil {
-			return nil, fmt.Errorf("error marshalling Float type length: %w", err)
+		intLength, tlvErr := marshalLength(len(intBytes))
+		if tlvErr != nil {
+			return nil, fmt.Errorf("error marshalling Float type length: %w", tlvErr)
 		}
 		intBuf.Write(intLength)
 		intBuf.Write(intBytes)
 
-		opaqueLength, err := marshalLength(len(intBuf.Bytes()))
-		if err != nil {
-			return nil, fmt.Errorf("error marshalling Float type length: %w", err)
+		opaqueLength, tlvErr := marshalLength(len(intBuf.Bytes()))
+		if tlvErr != nil {
+			return nil, fmt.Errorf("error marshalling Opaque length: %w", tlvErr)
+		}
+		if err = marshalTLV(tmpBuf, byte(ObjectIdentifier), oid); err != nil {
+			return nil, err
 		}
-		tmpBuf.Write([]byte{byte(ObjectIdentifier), byte(len(oid))})
-		tmpBuf.Write(oid)
 		tmpBuf.WriteByte(byte(Opaque))
 		tmpBuf.Write(opaqueLength)
 		tmpBuf.Write(intBuf.Bytes())
 
-		length, err := marshalLength(len(tmpBuf.Bytes()))
-		if err != nil {
-			return nil, fmt.Errorf("error marshalling Float type length: %w", err)
+		if err = marshalTLV(pduBuf, byte(Sequence), tmpBuf.Bytes()); err != nil {
+			return nil, err
 		}
 
-		// Sequence, length of oid + oid, then oid/oid data
-		pduBuf.WriteByte(byte(Sequence))
-		pduBuf.Write(length)
-		pduBuf.Write(tmpBuf.Bytes())
-
 	case Counter64:
-		tmpBuf.Write([]byte{byte(ObjectIdentifier), byte(len(oid))})
-		tmpBuf.Write(oid)
+		if err = marshalTLV(tmpBuf, byte(ObjectIdentifier), oid); err != nil {
+			return nil, err
+		}
 		tmpBuf.WriteByte(byte(pdu.Type))
-		intBytes, err := marshalUint64(pdu.Value)
-		if err != nil {
-			return nil, fmt.Errorf("error marshalling: %w", err)
+		intBytes, tlvErr := marshalUint64(pdu.Value)
+		if tlvErr != nil {
+			return nil, fmt.Errorf("error marshalling: %w", tlvErr)
```

Comment:
I'm not sure why this case needs a different error var. Also, this one isn't using the `marshalTLV` function, so why `tlvErr`?